internal/manifest/coi_iso_tree.go

```diff
@@ -66,6 +67,26 @@ func (p *CoreOSISOTree) serialize() osbuild.Pipeline {
 		osbuild.NewFilesInputs(osbuild.NewFilesInputReferencesPipeline(p.payloadPipeline.Name(), p.payloadPipeline.Filename)),
 	))
 
+	if p.coiPipeline.Ignition != nil {
+		cso := osbuild.CopyStageOptions{}
+		filename := ""
+		copyInput := ""
+		// These specific filenames in the root of the ISO are expected by
+		// coreos-installer-dracut during installation
+		if p.coiPipeline.Ignition.Data64 != "" {
+			filename = "ignition_config"
+			copyInput = p.coiPipeline.Ignition.Data64
+		} else {
+			filename = "ignition_url"
+			copyInput = p.coiPipeline.Ignition.ProvisioningURL
+		}
+		cso.Paths = append(cso.Paths, osbuild.CopyStagePath{
+			From: fmt.Sprintf("input://inlinefile/sha256:%x", sha256.Sum256([]byte(copyInput))),
+			To:   fmt.Sprintf("tree:///%s", filename),
+		})
```

Comment:
TINIEST of nitpicks, but I find the initialisation of `cso` and the `append` a bit strange here.  It could just be initialised at after the conditions with
```go
cso := osbuild.CopyStageOptions{
    Paths = ...
}
```
instead of appending.

Reply:
very true, I was over complicating things